Frontend/src/app/home/filter-dialog/filter-dialog.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import {
+	MAT_DIALOG_DATA,
+	MatDialogActions,
+	MatDialogContent,
+	MatDialogRef,
+	MatDialogTitle,
+} from '@angular/material/dialog';
+import { FilterDialogData } from '../../model/filter-dialog-data.model';
+import { MatActionList, MatList, MatListItem } from '@angular/material/list';
+import { MatFormField, MatOption, MatSelect } from '@angular/material/select';
+import {
+	FormBuilder,
+	FormControl,
+	FormGroup,
+	ReactiveFormsModule,
+} from '@angular/forms';
+import { MatButton } from '@angular/material/button';
+
+@Component({
+	selector: 'app-filter-dialog',
+	standalone: true,
+	imports: [
+		MatList,
+		MatSelect,
+		MatFormField,
+		MatOption,
+		ReactiveFormsModule,
+		MatDialogContent,
+		MatDialogTitle,
+		MatActionList,
+		MatListItem,
+		MatDialogActions,
+		MatButton,
+	],
+	templateUrl: './filter-dialog.component.html',
+	styleUrl: './filter-dialog.component.scss',
+})
+export class FilterDialogComponent implements OnInit {
+	protected formGroup: FormGroup;
+	protected priceCategoryControl = new FormControl('');
+	protected priceCategories: number[] = [];
+	protected foodCategories: string[] = [];
+
+	constructor(
+		public fb: FormBuilder,
+		public dialogRef: MatDialogRef<FilterDialogComponent>,
+		@Inject(MAT_DIALOG_DATA) public data: FilterDialogData
+	) {
+		this.formGroup = this.fb.group({
+			priceCategoryControl: [''],
+			foodCategoryControl: [''],
+		});
+	}
+
+	public ngOnInit(): void {
+		if (this.data) {
+			this.foodCategories = this.data.foodCategories;
+			this.priceCategories = this.data.priceCategories;
+			this.formGroup
+				.get('foodCategoryControl')
+				?.patchValue(this.data.selectedFoodCategories);
+			this.formGroup
+				.get('priceCategoryControl')
+				?.patchValue(this.data.selectedPriceCategories);
```

Comment:
I think you can 2 way bind the value property of the mat-select to `this.data.selectedFoodCategories`. To prevent selected data from overriding the original one you could make a copy of it and only updating the original if the form is submitted